posts/setting-up-a-shared-library-and-seed-job-in-jenkins-2.md

```diff
@@ -0,0 +1,189 @@
+    def testName = jobName + "_test"
+
+    createPipelineJob(deployName, repoUrl)
+    createMultibranchPipelineJob(testName, repoUrl)
+}
+
+buildPipelineJobs()
+```
+
+## Goal 3
+Since we will be setitng up all of our stages in a Shared Library, we need to set up a groovy script (`jenkinsJob.groovy`) that the `poc-micro` service points to from its Jenkinsfile. 
+
+### Adding `jenkinsJob.groovy` to `microservice-pipelines`
+We are going to set up our `jenkinsJob.groovy` to checkout our microservice code from source control and execute specific maven commands depending on the job that is running.
+  1. Check out our microservice repository.
+     * Check out the [workflow scm steps](https://jenkins.io/doc/pipeline/steps/workflow-scm-step/) for more information.
+  2. Read from the [`environment variables`](https://wiki.jenkins.io/display/JENKINS/Building+a+software+project#Buildingasoftwareproject-JenkinsSetEnvironmentVariables) (`env.JOB_NAME`) in order to obtain the Jenkins job name.
+     * `deployName = poc-micro_deploy = env.JOB_NAME`
+     * `testName = poc-micro_test = env.JOB_NAME`
+     * Check out `Jenkins Set Environment Variables` section at the [`Building a software project`](https://wiki.jenkins.io/display/JENKINS/Building+a+software+project#Buildingasoftwareproject-JenkinsSetEnvironmentVariables) Jenkins wiki for more information on `env.JOB_NAME`.
+
+Create and add the below code to `vars/jenkinsJob.groovy` in the `part2` branch of `microservice-pipelines`.
+```groovy
+def call(){
+    node {
+        stage('Checkout') {
+            checkout scm
+        }
+
+        // Execute different stages depending on the job
+        if(env.JOB_NAME.contains("deploy")){
+            packageArtifact()
+        } else if(env.JOB_NAME.contains("test")) {
+            buildAndTest()
+        }
+    }
+}
+
+def packageArtifact(){
+    stage("Package artifact") {
+        sh "mvn package"
+    }
+}
+
+def buildAndTest(){
+    stage("Backend tests"){
+        sh "mvn test"
+    }
+}
+```
+
+## Goal 4
+In order for our microservices to execute in Jenkins, we need a `Jenkinsfile` in `poc-micro`. 
+. 
+### Setting up our `Jenkinsfile` in our microservice
+We will configure a `Jenkinsfile` in our microservices to point to our `mciroservice-piplines` Shared Library. 
+   * **Note** We are introducing a the `version specifier` feature associated with the `@Library` annotation in Shared Libraries here. Within the `@Library` annotation, you will always need to provide the name of your Shared Library (e.g. `@Library("microservice-pipelines)`). However, if you add another `@` sign at the end of the Shared Library name (`microservive-pipelines`), you can tell your `Jenkinsfile` to read specific branches or tags from your Shared Library. In fact, in the code below, that is what we did. We are signaling our `Jenkinsfile` to use the `part2` branch of our Shared Library with `@part2` (e.g. `@Library("microservice-pipelines@part2")`).
```

Comment:
a the - typo

Reply:
The latest version has this and other type-o's corrected